codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2HeadersDecoder.java

```diff
@@ -126,4 +126,28 @@ public Http2Headers decodeHeaders(int streamId, ByteBuf headerBlock) throws Http
             throw connectionError(COMPRESSION_ERROR, e, e.getMessage());
         }
     }
+
+    /**
+     * A weighted moving average estimating how many headers are expected during the decode process.
+     * @return an estimate of how many headers are expected during the decode process.
+     */
+    protected final int numberOfHeadersGuess() {
+        return (int) headerArraySizeAccumulator;
+    }
+
+    /**
+     * Determine if the headers should be validated as a result of the decode operation.
+     * @return {@code true} if the headers should be validated as a result of the decode operation.
+     */
+    protected final boolean validateHeaders() {
+        return validateHeaders;
+    }
+
+    /**
+     * Create a new {@link Http2Headers} object which will store the results of the decode operation.
+     * @return a new {@link Http2Headers} object which will store the results of the decode operation.
+     */
```

Comment:
I'm not necessarily against this but lets consider adding that dependency in a followup PR to keep the change isolated.